src/spi.rs

```diff
@@ -266,29 +350,40 @@ macro_rules! spi {
                         w.mbr()
                             .variant(mbr) // master baud rate
                             .dsize()
-                            .bits(8 - 1) // 8 bit frames
+                            .bits(config.frame_size - 1)
                     });
 
                     // ssi: select slave = master mode
                     spi.cr1.write(|w| w.ssi().slave_not_selected());
 
+                    // Calculate the CS->transaction cycle delay bits.
+                    let mut cycle_delay = (config.cs_delay * spi_freq as f32) as u32;
```

Comment:
This could be a good place to use [`ceil`](https://doc.rust-lang.org/std/primitive.f32.html#method.ceil). So that we promise a delay that equal or longer than specified, but never shorter.

Reply:
`ceil` appears to be part of `std::f32` as opposed to `core::f32`, so I don't think we can pull it in

Reply:
I would also advocate to `ceil`/`floor` all the cases where one typically supplies a datasheet min/max value and can reasonably expect the hal to honor that limit.
Without math in `core`, and the numeric cast being truncation we could still implement a dirty `ceil` here.

Reply:
Agreed - I don't think we should silently ignore the truncation (although forgot to indicate this), since the intended use case here is to adhere to some minimum hold time.

Reply:
A simple (dirty) workaround I have implemented is to add 1 if the cs_delay is specified as non-zero, which will ensure that we always have at least the number of clock cycles necessary (up to 15).